src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+// #[PerformanceCriticalPath]
+use txn_types::{Key, TimeStamp};
+
+use crate::storage::{
+    mvcc::MvccReader,
+    txn::{
+        commands::{
+            Command, CommandExt, FlashbackToVersion, ProcessResult, ReadCommand, TypedCommand,
+            FLASHBACK_BATCH_SIZE,
+        },
+        sched_pool::tls_collect_keyread_histogram_vec,
+        Result,
+    },
+    ScanMode, Snapshot, Statistics,
+};
+
+command! {
+    FlashbackToVersionReadPhase:
+        cmd_ty => (),
+        display => "kv::command::flashback_to_version_read_phase | {:?}", (ctx),
+        content => {
+            version: TimeStamp,
+            end_key: Option<Key>,
+            next_lock_key: Option<Key>,
+            next_write_key: Option<Key>,
+        }
+}
+
+impl CommandExt for FlashbackToVersionReadPhase {
+    ctx!();
+    tag!(flashback_to_version);
+    request_type!(KvFlashbackToVersion);
+    property!(readonly);
+    gen_lock!(empty);
+
+    fn write_bytes(&self) -> usize {
+        0
+    }
+}
+
+impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
```

Comment:
why need read phase

Reply:
I took the design of `ResolveLock`/`ResolveLockReadPhase` and divide this operation, which requires reading before writing, into two stages. I think the advantage of this is that it is more in line with the design model of `txn::Command`, so that we don't have to write some special logic because we can't determine the keys involved in `FlashbackToVersion` to bypass the latch lock fetching.